packages/teams-js/src/internal/communication.ts

```diff
@@ -257,6 +264,90 @@ export function sendMessageToParentAsync<T>(
   });
 }
 
+type SimpleType = string | number | boolean | null | undefined | SimpleType[];
+
+function serializeItemArray(items: (SimpleType | ISerializable)[]): (SimpleType | object)[] {
+  return items.map((item) => {
+    if (isSerializable(item)) {
+      return item.serialize();
+    } else {
+      return item;
+    }
+  });
+}
+
+/**
+ * Call a function in the host and receive a response. If the host returns an {@link SdkError} instead of a normal response, this function will throw a new Error containing the SdkError's information
+ *
+ * @param functionName The function name to call in the host.
+ * @param args A collection of data to pass to the host. This data must be an array of either simple types or objects that implement {@link ISerializable}.
+ * @param responseHandler When the host responds, this handler will validate and deserialize the response.
+ * @param apiVersionTag A unique tag used to identify the API version for telemetry purposes. This should be set using {@link getApiVersionTag}, which should be passed a unique string identifying the function being called by the app developer as well as a version number that is incremented whenever meaningful changes are made to that function.
+ * @param isResponseAReportableError This optional property can be used to override the default ErrorChecking this function uses to decide whether to throw the host response as a new Error. Specify this is your function needs to do any logic verifying that the object received is an error that goes beyond the logic found in {@link isSdkError}.
```

Comment:
>logic found in {@link isSdkError [](http://example.com/codeflow?start=310&length=32)

I'm not... actually sure if it matters to update this but this function isn't using `isSdkError` anymore. Writing this documentation to be 100% accurate may not be necessary, though.

Reply:
I think everything I wrote in here is accurate: the host returns SdkError objects and we turn them into Errors that contain the SdkError information. Let me know if anything in there is confusing or you want more detail